python/pylibcudf/pylibcudf/io/types.pyx

```diff
@@ -269,6 +282,14 @@ cdef class TableInputMetadata:
     """
     def __init__(self, Table table):
         self.c_obj = table_input_metadata(table.view())
+        self.column_metadata = []
+        self.table = table
```

Comment:
*question*: Why are we holding on to the `table` here? the C-level object doesn't hold on to references to the `table_view`.

Reply:
I thought it was maybe a nice-to-have. I removed for now, since its un.needed